pkg/tests/ossm/setup.go

```diff
@@ -81,7 +86,15 @@ func BasicSetup(t test.TestHelper) {
 func DeployControlPlane(t test.TestHelper) {
 	t.T().Helper()
 	t.LogStep("Apply default SMCP and SMMR manifests")
-	InstallSMCP(t, meshNamespace)
+	smcpValues := DefaultSMCP()
+	clusterWideProxy := oc.GetProxy(t)
+	if clusterWideProxy != nil {
+		smcpValues.ClusterWideProxy = true
+		smcpValues.HttpProxy = clusterWideProxy.HTTPProxy
+		smcpValues.HttpsProxy = clusterWideProxy.HTTPSProxy
+		smcpValues.NoProxy = clusterWideProxy.NoProxy
+	}
```

Comment:
Why not add this in `DefaultSMCP()` so that it's always used? Some tests don't call `DeployControlPlane`, but call `InstallSMCP()`. Don't those tests fail on ipv6?

Reply:
Not knowing test structure I'm +1 to always set clusterwide proxy if it's available. Customers will probably never run control plane without this setting if there is a cluster wide proxy configured in their cluster.

Reply:
> Why not add this in DefaultSMCP() so that it's always used?

I'm working on a specific issue and I'm focused on solving issues one by one. It may be refactored later by @fjglira.

> Don't those tests fail on ipv6?

No, as far as istiod will not connect to external services. I don't know if there are other cases where istiod connects to external services.

Reply:
But still... moving this code to DefaultSMCP() takes 10s, and it's the better place for this code.

Reply:
No, it's not 10 seconds. I would have to refactor a lot of code and verify it that works. To make that work, I would have to add `t test.TestHelper` as an argument to `DefaultSMCP()`. In some cases, like this [one](https://github.com/maistra/maistra-test-tool/blob/main/pkg/tests/tasks/traffic/egress/yaml_vars.go#L30), I don't have `TestHelper` available, so I would have to increase the scope of this PR and spend time on refactoring the code base. I don't have enough time right now and I want to just fix issues. It can be refactored after the release.